generation/generate_release_please_config.sh

```diff
@@ -1,70 +1,60 @@
+  module_name="${path:2}"
+  version_file="${path}/versions.txt"
 
-  module_name="${module:2}"
-
-  module_released_version=$(grep google- ${version_file} |head -1 |awk -F: '{print $2}')
-  module_snapshot_version=$(grep google- ${version_file} |head -1 |awk -F: '{print $3}')
-  artifactName_config=$(grep google- ${version_file} |head -1 |awk -F: '{print $1}')
-
-  #specific case
-  if [[ ${module_name} == "java-grafeas" ]]; then
-    module_released_version=$(grep grafeas ${version_file} |head -1 |awk -F: '{print $2}')
-    artifactName_config=$(grep grafeas ${version_file} |head -1 |awk -F: '{print $1}')
+  # Java-Grafeas is a special case
+  if [[ "${module_name}" = "java-grafeas" ]]; then
+    module_line=$(grep -E "^grafeas:[0-9]+\.[0-9]+\.[0-9]+.*:[0-9]+\.[0-9]+\.[0-9]+.*$" "${version_file}")
+  else
+    module_line=$(grep -E "^google-.*:[0-9]+\.[0-9]+\.[0-9]+.*:[0-9]+\.[0-9]+\.[0-9]+.*$" "${version_file}")
   fi
+  artifact_name=$(echo "${module_line}" | cut -d ":" -f1)
+  module_released_version=$(echo "${module_line}" | cut -d ":" -f2)
+  module_snapshot_version=$(echo "${module_line}" | cut -d ":" -f3)
 
-  #concatenating module name and module version
-  rp_manifest_line=""\""${module_name}"\"": "\""${module_released_version}"\"""
+  # concatenating module name and module version
+  rp_manifest_line="${tab}\"${module_name}\": \"${module_released_version}\""
+  # Generate the JSON block with formatting
+  rp_config_line+="${tab}${tab}\"${module_name}\": {\n${tab}${tab}${tab}\"component\": \"${artifact_name}\",\n${tab}${tab}${tab}\"skip-github-release\": true\n${tab}${tab}}"
 
-  rp_config_line+=""\""${module_name}"\"": {\n\
-          "\""component"\"": ""\""${artifactName_config}"\"","\n\
-          "\""skip-github-release"\"": "true"\n\
-         }"
-
-  #adding " , " where it's necessary
-  if [[ ${num_modules} -gt 2 ]]; then
+  # adding " , " where it's necessary
+  if [[ ${num_modules} -gt 1 ]]; then
     rp_manifest_line+=","
-    rp_config_line+=",\n    "
-    num_modules=$((num_modules-1))
+    rp_config_line+=",\n"
+    num_modules=$((num_modules - 1))
   fi
 
-  #adding the line to manifest config file
-  echo "${rp_manifest_line}" >> .release-please-manifest.json
-
-  #for multi-module libraries, sync the versions to released(non-snapshot):current(snapshot) type
-  cat ${version_file} | while read line
-      do
-         if [[ ${line} =~ [0-9] ]] && ! [[ ${line} == *"SNAPSHOT"* ]] ; then
-           artifact_name=$(echo "${line}" | awk -F':' '{print $1}')
-           old_version=$(echo "${line}" | awk -F':' '{print $3}')
-           new_version=$(echo ${old_version} |  awk -F'.' '{print $1"."$2"."$3+1}' |  sed s/[.]$//)
-           new_version="${new_version}-SNAPSHOT"
-           sed -i.bak -e "s|${artifact_name}:${old_version}:${old_version}|${artifact_name}:${old_version}:${new_version}|" "${version_file}" && rm "${version_file}".bak
-         fi
-      done
```

Comment:
Is this removed because we will use `./generation/update_versions.sh`? (Good)

Reply:
Yep.